llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py

```diff
@@ -183,43 +184,92 @@ def __init__(
     def class_name(cls) -> str:
         return "HuggingFaceEmbedding"
 
-    def _embed(
+    def _validate_input(self, text: str) -> None:
+        """Validate input text.
+
+        Args:
+            text: Input text to validate
+
+        Raises:
+            ValueError: If text is empty or exceeds max length
+        """
+        if not text or not text.strip():
+            raise ValueError("Input text cannot be empty or whitespace")
+        if len(text) > self.max_length:
+            raise ValueError(
+                f"Input text length {len(text)} exceeds maximum {self.max_length}"
```

Comment:
My guess is this will cause a LOT of errors for users if merged. Previously, the default behaviour was to truncate if the text is too long.

Reply:
You raised a good point about the behaviour change related to max length validation. I have removed it as the max length validation was redundant.